### PR TITLE
Refactor intltool related code, and add proper testing

### PR DESCRIPTION
**Pull request by stefwalter**
We're not a standard intltool project, so lets just focus on
the stuff we use and need:

* [x] Cleanup use of translations in C code
* [x] Install mo files in LC_MESSAGES
* [x] Install po.xx.js files
* [x] Only extract relevant strings into js and mo files
* [x] Have a standard update-po target ... that also works in the po directory.
* [x] Detect unmarked translations in 'make check' (not needed)
* [x] Have real tests for localization.

We won't cover things like translatable attributes in our update-po and check targets yet.

### CODE REVIEW

**Comment by dperpeet** [comment]
needs rebase

**Comment by stefwalter** [comment]
Rebase resolved automatically.

File: pkg/playground/react-demo-dialog.jsx [hunk 1]
```diff
@@ -40,7 +40,7 @@
                         <tr>
                             <td className="top">
                                 <label className="control-label" for="control-1">
-                                    {_("Label")}
+                                    {"Label"}
```
Comment (mvollmer) [inline pkg/playground/react-demo-dialog.jsx]: If this shouldn't be translated, then we can just write ```<label ...>Label</label>```, no?

Comment (mvollmer) [inline pkg/playground/react-demo-dialog.jsx]: For your convenience :-) https://github.com/mvollmer/cockpit/commit/bfedddec918c8c1bc4f2772572aa75de9270c30b

Comment (stefwalter) [inline pkg/playground/react-demo-dialog.jsx]: Thanks. Squashed into the relevant commit.

**Comment by petervo** [comment]
Seems jsx files may not be getting processed. I'm not seeing strings from the selinux jsx files in the generated po files.

**Comment by stefwalter** [comment]
Added missing .jsx and .es6  to ```make update-po``` target and po files.

**Comment by mvollmer** [comment]
We see this for some rhel-atomic, fedora-25, and debian-unstable:

```
make: *** No rule to make target 'doc/guide/html', needed by 'distdir'.  Stop.
```

(No idea what is common here, maybe master was different for those test runs.)

**Comment by mvollmer** [comment]
> (No idea what is common here, maybe master was different for those test runs.)

They all ran on dragon.

**Comment by stefwalter** [comment]
I don't think those failures have to do with this pull request. They must have to do with #5425 and using the same builddir when testing one or the other.

**Pull request merged**